File: weather-demo-app-android/app/src/main/java/com/demo/weather/ui/welcome/WelcomeScreen.kt
package com.demo.weather.ui.welcome

import androidx.compose.runtime.Composable

@Composable
fun WelcomeScreen(
    viewModel: WelcomeViewModel,
    onContinue: () -> Unit
) {
    WelcomeView(
        onContinueClicked = {
            viewModel.continueToApp(onFinished = onContinue)
        }
    )
}

File: weather-demo-app-apple/Weather/UI/Welcome/WelcomeScreen.swift
//
//  WelcomeScreen.swift
//  Weather
//
//  Created by Onur on 4/11/26.
//

import SwiftUI

struct WelcomeScreen: View {
    @ObservedObject var viewModel: WelcomeViewModel
    let onContinue: () -> Void

    var body: some View {
        WelcomeView(
            onContinueClicked: {
                viewModel.continueToApp(onFinished: onContinue)
            }
        )
    }
}
